### PR TITLE
refines project phases activity log events

### PR DESCRIPTION
**Pull request by tillyw**
this pr refines phases and subphase activity log events using REST connectors

## Associated issues
fixes https://github.com/cityofaustin/atd-data-tech/issues/11279

## Testing
**URL to test:** 
https://deploy-preview-953--atd-moped-main.netlify.app/moped/

**Steps to test:**
- add, edit and remove a phase with a subphase and a phase without a subphase while noting the activity log updates
- the changes should be reflected in the activity log with the subphase listed where appropriate

Create event
```markdown
Added **<phase-name> - <subphase-name-if-exists>** as a new phase
```
Edit event

```markdown
Edited the phase **<phase-name> - <subphase-name-if-exists>** by updating the <comma-separated-field names>
```

Delete event 
```markdown
Removed the phase **<phase-name> - <subphase-name-if-exists>**
```

---
#### Ship list
- [ ] Code reviewed 
- [ ] Product manager approved
- [ ] Added to [QA test script](https://docs.google.com/spreadsheets/d/1n_O6MLh9cwwPf57HUM394Ea-z9uuoEV1-QW4axNZXLE/edit#) if applicable

### CODE REVIEW

**Review by johnclary — CHANGES_REQUESTED** [review]
woohoo! this looks great! i'm so happy to see these phase entries cleaned up.

one tiny change—would you rename the subphase label from **subphase ID** to just **subphase**?

Screen Shot 2023-01-25 at 11 42 20 AM

**Comment by tillyw** [comment]
> woohoo! this looks great! i'm so happy to see these phase entries cleaned up.
> 
> one tiny change—would you rename the subphase label from **subphase ID** to just **subphase**?
> 
> Screen Shot 2023-01-25 at 11 42 20 AM

thanks, john! i as you requested and cleaned up the code a bit more to make it more readable and easier to edit - also got rid of a bunch of unnecessary entries in `ProjectActivityLogTableMaps`

**Review by mddilley — APPROVED** [review]
Yayyy! Love seeing the web of the activity log grow. 🕸️  Fun to see the new pattern handling a field that it isn't always there too. ✨ All test steps work as promised. 🚢

File: moped-editor/src/utils/activityLogFormatters/mopedPhasesActivity.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+      changeText: [
+        { text: "Deleted the phase ", style: null },
+        phaseText,
+        // include subphase name if one exists
+        ...(subphase ? [subphaseText] : []),
+      ],
+    };
+  }
+
+  // Multiple fields in the moped_proj_funding table can be updated at once
```
Comment (chiaberry) [inline moped-editor/src/utils/activityLogFormatters/mopedPhasesActivity.js:46]: moped_proj_funding --> moped_proj_phases

File: moped-editor/src/utils/activityLogFormatters/mopedPhasesActivity.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+  // add a new phase
+  if (change.description.length === 0) {
+    return {
+      changeIcon,
+      changeText: [
+        { text: "Added ", style: null },
+        phaseText,
+        // include subphase name if one exists
+        ...(subphase ? [subphaseText] : []),
```
Comment (chiaberry) [inline moped-editor/src/utils/activityLogFormatters/mopedPhasesActivity.js:27]: ✨ i like this

**Review by chiaberry — APPROVED** [review]
its exciting to see these get knocked out

**Review by johnclary — APPROVED** [review]
🚢 🚢 🚢 🚢 🚢 🚢 🚢  🙌 💯 📆

File: moped-editor/src/utils/activityLogFormatters/mopedPhasesActivity.js (hunk 1 above)
Comment (tillyw) [reply moped-editor/src/utils/activityLogFormatters/mopedPhasesActivity.js:46]: going to merge this but i will fix this typo in another pr!

**Comment by roseeichelmann** [comment]
Oops got to this a few minutes late but this looks great! My only comment was that the activity log for adding a new phase has a period and the others don't, so maybe the period could be nixxed for continuity between them :)

Screenshot from 2023-01-26 16-05-23

**Comment by tillyw** [comment]
> Oops got to this a few minutes late but this looks great! My only comment was that the activity log for adding a new phase has a period and the others don't, so maybe the period could be nixxed for continuity between them :)
> 
> Screenshot from 2023-01-26 16-05-23

thanks @roseeichelmann !! i'm fixing that in one of the follow-up activity log prs!

**Pull request merged**